aiida/orm/implementation/sqlalchemy/code.py

```diff
@@ -130,7 +130,8 @@ def delete_code(code):
             "has {} output links".format(len(existing_outputs)))
     else:
         repo_folder = code._repository_folder
-        from aiida.backends.sqlalchemy import session
+        from aiida.backends.sqlalchemy import get_scoped_session
+        session = get_scoped_session()
         session.begin(subtransactions=True)
         try:
             code.dbnode.delete()
```

Comment:
This has nothing to do with any issues now, but shouldn't the order of folder.erase and session.commit be reversed?
I.e. only erase a folder if the commit went through?